playwright/src/main/java/com/microsoft/playwright/impl/ChannelOwner.java

```diff
@@ -69,9 +69,18 @@ void disconnect() {
   }
 
   <T> T withWaitLogging(String apiName, Supplier<T> code) {
-    return super.withLogging(apiName, new WaitForEventLogger<>(this, apiName, code));
+    return withLogging(apiName, new WaitForEventLogger<>(this, apiName, code));
   }
 
+  @Override
+  <T> T withLogging(String apiName, Supplier<T> code) {
+    String previousApiName = connection.setApiName(apiName);
+    try {
+      return super.withLogging(apiName, code);
+    } finally {
+      connection.setApiName(previousApiName);
```

Comment:
having it on the connection feels somehow wrong and racy. But not familiar with pw-Java.

Reply:
Connection is required to be used with exclusive access and hence it guarantees here that we'll not have a race. I agree thought that Connection is somewhat abused for this purpose, we can refactor this if it becomes an issue.